### ISSUE TITLE
Bug: Prisma crashes when connecting to CockroachDB with "Segmentation fault" error

### ISSUE THREAD

**Issue by peterkibuchi** [t3-oss/create-t3-app#1308]
### Provide environment information

**Binaries:**
  Node: 18.15.0
  pnpm: 8.0.0

**Project Specifics:**
create-t3-app initVersion: 7.8.0
DB: CockroachDB

### Describe the bug

I was following along with [Theo's T3 Tutorial](https://youtu.be/YkOSUVzOAA4) when around 30 minutes in my dev server stopped (crashed) unexpectedly; no error was displayed. I tried restarting it a bunch, but every time it would build the app and crash; no error/information was displayed:
```bash
$ pnpm dev

> michelangelo@0.1.0 dev ~/Development/michelangelo
> next dev

ready - started server on 0.0.0.0:3000, url: http://localhost:3000
info  - Loaded env from ~/Development/michelangelo/.env
event - compiled client and server successfully in 934 ms (310 modules)
wait  - compiling /api/trpc/[trpc] (client and server)...
event - compiled successfully in 114 ms (61 modules)
wait  - compiling /src/middleware (client and server)...
event - compiled successfully in 133 ms (81 modules)
[crashes with no error]
```
The same thing happened when I ran `prisma studio`.

I tried everything: from upgrading dependencies, to upgrading pnpm, and even node to the latest versions; nothing worked. I did however get the following error from pnpm (after upgrading it to `v8.0.0`) when I ran Prisma Studio: `ERR_PNPM_RECURSIVE_EXEC_FIRST_FAIL  Command was killed with SIGSEGV (Segmentation fault): prisma studio`

A quick search revealed that this is [a known issue](https://github.com/prisma/prisma/issues/14122) within Prisma; it seems that Prisma doesn't play well with Postgres and Postgres-like DBs (e.g. Supabase and CockroachDB) within `Node v18` environments. The issue (dev server crashing continuously) goes away [when one reverts to Node v16](https://github.com/prisma/prisma/issues/14122#issuecomment-1245368481) (or at least it did for me). [Here](https://github.com/prisma/prisma/issues/14122#issuecomment-1249205365) are alternative ways to solve the problem.

I created this issue to help those who stumble across the same issue but have no idea where to start, since the dev server crashes with no warnings/errors. I would also like to request that this quirk of Prisma be added to the docs for people using Postgres DBs to know of it beforehand.

### Reproduction repo

N/A

### To reproduce

1. Create T3 app with create-t3-app using pnpm in an `Node v18 (lts/hydrogen)` environment.
2. Provision either a Supabase Postgres DB or CockroachDB.
3. Run dev server. You should run into this issue after some time.

**Comment by zanzlender** [comment]
I have a similar problem #1342.

But i believe it's due to NodeJS. Can you try to build and run the same app only on Node 16x?

**Comment by based64-eth** [comment]
are you using WSL by any chance?

**Comment by peterkibuchi** [comment]
No.

**Comment by juliusmarminge** [comment]
Has this been fixed upstream yet? They closed the issue.

Can anyone verify?

**Comment by peterkibuchi** [comment]
Hi Julius. I don't know whether they fixed the issue since I reverted to Node 16. Bit held up atm, but will verify and get back to you within the week.

**Comment by peterkibuchi** [comment]
I have tested Prisma in a Node 18.16.0 (lts/hydrogen) environment against both a Supabase Postgres DB and CockroachDB. Can confirm that the error no longer occurs. [This post](https://github.com/prisma/prisma/issues/14122#issuecomment-1511429093) shows that it was fixed with the release of Prisma 4.13.0. Closing this now.
